src/main/java/pl/jkap/sozzt/contract/domain/ContractFacade.java

```diff
@@ -1,55 +1,60 @@
     private final ContractRepository contractRepository;
-    private final ContractCreator contractCreator;
+    private final ContractMapper contractMapper;
 
-    public ContractFacade(ContractRepository contractRepository, ContractCreator contractCreator) {
-        this.contractCreator = contractCreator;
+    public ContractFacade(ContractRepository contractRepository, ContractMapper contractMapper) {
+        this.contractMapper = contractMapper;
         this.contractRepository = contractRepository;
     }
 
     public ContractDto addContract(ContractDto contractDTO) {
         requireNonNull(contractDTO);
-        Contract contract = contractCreator.from(contractDTO);
-        contract = contractRepository.save(contract);
-        return contract.dto();
+        ContractEntity contractEntity = contractMapper.from(contractDTO);
+        DataInputContract dataInputContract = contractMapper.dataInputStepFrom(contractEntity);
+        return contractRepository.save(dataInputContract.toContractEntity()).dto();
     }
 
     public ContractDto getContract(long id) {
-        return contractRepository.findById(id).dto();
+        return contractRepository.findById(id).orElseThrow(ContractNotFoundException::new).dto();
     }
 
     @EventListener
-    public void uploadedScanFromTauron(FileUploadedSpringEvent event) {
-        confirmScanUploaded(event.getMessage());
-    }
-
-    public ContractDto confirmStep(long idContract) {
-        Contract contract = contractRepository.findById(idContract);
-        contract.confirmStep();
-        return contractRepository.save(contract).dto();
+    public void uploadedScanFromTauron(FileUploadedSpringEvent fileUploadedSpringEvent) {
+        confirmScanUploaded(fileUploadedSpringEvent.getIdContract());
     }
 
     public void confirmScanUploaded(long idContract) {
-        Contract contract = contractRepository.findById(idContract);
-        contract.confirmScanUploaded();
-        contractRepository.save(contract);
-
+        ContractEntity contractEntity = contractRepository.findById(idContract).orElseThrow(ContractNotFoundException::new);
+        DataInputContract dataInputContract = contractMapper.dataInputStepFrom(contractEntity);
+        dataInputContract.setScanFromTauronUpload(true);
+        contractRepository.save(dataInputContract.toContractEntity());
     }
-    public boolean checkIsScanFromTauronUploaded(long idContract){
-        Contract contract = contractRepository.findById(idContract);
-        return contract.checkIsScanFromTauronUploaded();
-    }
-
 
+    public ContractDto confirmStep(long idContract) {
+        ContractEntity contractEntity = contractRepository.findById(idContract).orElseThrow(ContractNotFoundException::new);
+        Contract contract = contractMapper.from(contractEntity);
+        Contract confirmContract = contract.confirmStep();
+        contractEntity.setContractStepEnum(confirmContract.toContractEntity().getContractStepEnum());
```

Comment:
you shouldn't convert whole object just to set one enum in entity, domain object `Contract`` should give you this information